graph/patterns/change-tracking.md

```diff
@@ -0,0 +1,163 @@
+  - There is no mechanism to indicate that a resource has entered or exited the dataset based on a change that causes it to match or no longer match any `$filter` query parameter.
+  - When a link to an entity is deleted, when the linked entity is deleted, or when a link to an entity is added, the implementer MUST return a `property@delta` annotation. 
+  - When a link to an entity is deleted, but the entity still exists, the reason MUST be set to `changed`.
+  - When a link to an entity is deleted along with the entity, the reason MUST be set to `deleted`.
+  
+API producers MAY choose to collate multiple changes to the same resource into a single change record. 
+
+## When to use this pattern
+
+API consumers want a pull mechanism to request and process change to Microsoft Graph data, either via proactive polling or by responding to Microsoft Graph notifications.
+
+API consumers need guaranteed data integrity over the set of changes to Microsoft Graph data.
+
+## Considerations
+
+ - API service MAY be able to respond to standard OData query parameters with the initial call to the delta function:
+
+    - `$select` to enforce the set of properties on which change is reported.
+    - `$filter` to influence the scope of changes returned.
+    - `$expand` to include linked resources with the set of changes.
+    - `$top` parameter to influence the size of the set of change records.
+  
+    These query parameters MUST be encoded into subsequent `@odata.nextLink` or `@odata.deltaLink`, such that the same options are preserved through the call sequence without callers respecifying them, which MUST NOT be allowed. OData query parameters must be honored in full, or a 400-error returned.
+- Making a sequence of calls to a delta function followed by the opaque URLs in the `nextLink` and `deltaLink` MUST guarantee that the data at the start time of the call sequence and all changes to the data thereafter will be returned at least once. It is not necessary to avoid duplicates in the sequence. When the delta function is returning changes, they MUST be sequenced chronologically refer to [public documentation](https://learn.microsoft.com/en-us/graph/delta-query-overview?view=graph-rest-1.0) for more details.
+- The delta function can be bound to a collection, as with
+`/users/delta` that returns the changes to the users' collection
+  or to some logical parent resource, where the change records are implied to be relative to all collections contained within the parent, for example
+  `/me/planner/all/delta` – this returns changes to any resource within planner that a user is subscribed to as a heterogenous collection.
+- API producers might use `$skipToken` and `$deltaToken` within their implementations of `nextLink` and `deltaLink`, however the URLs are defined as being opaque and the existence of the tokens MUST NOT be documented.   It is not a breaking change to modify the structure of `nextLinks` or `deltaLinks`.- 
```

Comment:
I'm not sure if this is true.  AGS looks for and encodes/decodes the skip and delta tokens. I think this is a *should/must* rather than *might*.